### PR TITLE
Impro1834 remove time zone

### PR DESCRIPTION
**Pull request by mspelman07**
Addresses #1834 

Removes code related to local time zones.
Acceptance tast data on branch impro1834_remove_time_zone

Testing:
 - [x] Ran tests and they passed OK

### CODE REVIEW

**Review by bayliffe — CHANGES_REQUESTED** [review]
Thanks Marcus, this looks good. There are still some bits to tidy up in: these files:
- improver/blending/weighted_blend.py
- improver_tests/blending/weighted_blend/test_WeightedBlendAcrossWholeDimension.py
- improver/developer_tools/metadata_interpreter.py
- improver_tests/developer_tools/test_MOMetadataInterpreter.py
- improver_tests/developer_tools/conftest.py (associated with above)
- improver/metadata/constants/time_types.py
- improver_tests/spotdata/test_SpotExtraction.py
- improver_tests/spotdata/test_build_spotdata_cube.py (modify the comment, retain the test)
- improver_tests/acceptance/test_spot_extract.py (some more acceptance test data to delete from spot-extract).

File: improver/developer_tools/metadata_interpreter.py [hunk 1]
```diff
@@ -92,7 +91,7 @@
 SPOT_COORDS = ["spot_index", "latitude", "longitude", "altitude", "wmo_id"]
 UNBLENDED_TIME_COORDS = ["time", "forecast_period", "forecast_reference_time"]
 BLENDED_TIME_COORDS = ["time", "blend_time"]
-LOCAL_TIME_COORDS = ["time", "time_in_local_timezone"]
+LOCAL_TIME_COORDS = ["time"]
```
Comment (bayliffe) [inline improver/developer_tools/metadata_interpreter.py:94]: You can delete `LOCAL_TIME_COORDS` entirely I believe.

File: improver_tests/spotdata/test_SpotExtraction.py [hunk 2]
```diff
@@ -105,36 +101,28 @@ def setUp(self):
...
-                    _create_time_point(time - timedelta(hours=1)),
-                    _create_time_point(time),
-                ]
-                for time in times
-            ]
-        )
+        location_points = np.array([location for location in locations])
+        bounds = np.array([[location - 1, location] for location in locations])
+
         # Broadcast the times to a 2-dimensional grid that matches the diagnostic
```
Comment (bayliffe) [inline improver_tests/spotdata/test_SpotExtraction.py:110]:
```suggestion
        # Broadcast the locations to a 2-dimensional grid that matches the diagnostic
```

File: improver_tests/spotdata/test_SpotExtraction.py [hunk 3]
```diff
@@ -105,36 +101,28 @@ def setUp(self):
...
-            time_points, "time", bounds=bounds, units=TIME_COORDS["time"].units
+        # Create a 2-dimensional auxiliary coordinate
+        self.location_aux_coord = iris.coords.AuxCoord(
+            location_points, long_name="location", bounds=bounds, units="degrees"
         )
 
         diagnostic_cube_2d_time = diagnostic_cube_yx.copy()
-        diagnostic_cube_2d_time.coord("time").rename("time_in_local_timezone")
-        diagnostic_cube_2d_time.add_aux_coord(self.time_aux_coord, data_dims=(0, 1))
+        diagnostic_cube_2d_time.add_aux_coord(self.location_aux_coord, data_dims=(0, 1))
```
Comment (bayliffe) [inline improver_tests/spotdata/test_SpotExtraction.py:120]:
```suggestion
        diagnostic_cube_2d_aux.add_aux_coord(self.location_aux_coord, data_dims=(0, 1))
```
Here and surrounds.

File: improver_tests/spotdata/test_SpotExtraction.py [hunk 4]
```diff
@@ -105,36 +101,28 @@ def setUp(self):
...
 
         expected_indices = [[0, 0], [0, 0], [2, 2], [2, 2]]
-        points = [self.time_aux_coord.points[y, x] for y, x in expected_indices]
-        bounds = [self.time_aux_coord.bounds[y, x] for y, x in expected_indices]
-        self.expected_spot_time_coord = self.time_aux_coord.copy(
+        points = [self.location_aux_coord.points[y, x] for y, x in expected_indices]
+        bounds = [self.location_aux_coord.bounds[y, x] for y, x in expected_indices]
+        self.expected_spot_time_coord = self.location_aux_coord.copy(
```
Comment (bayliffe) [inline improver_tests/spotdata/test_SpotExtraction.py:125]:
```suggestion
        self.expected_spot_aux_coord = self.location_aux_coord.copy(
```
And where it is used elsewhere in this file.

File: improver_tests/spotdata/test_SpotExtraction.py [hunk 5]
```diff
@@ -293,15 +281,16 @@ def test_scalar_and_nonscalar_coords(self):
...
+            if coord.name() in ["time", "forecast_reference_time", "forecast_period"]
         ]
         expected_nonscalar = [self.expected_spot_time_coord]
+
         x_indices, y_indices = self.coordinate_cube.data
 
         scalar, nonscalar = plugin.get_aux_coords(
             self.diagnostic_cube_2d_time, x_indices, y_indices
         )
+        print(nonscalar)
```
Comment (bayliffe) [inline improver_tests/spotdata/test_SpotExtraction.py:293]: Delete

File: improver_tests/spotdata/test_SpotExtraction.py [hunk 6]
```diff
@@ -311,7 +300,7 @@ def test_multiple_nonscalar_coords(self):
         of the 2D non-scalar input coordinates at spot sites."""
         plugin = SpotExtraction()
 
-        additional_2d_crd = self.time_aux_coord.copy()
+        additional_2d_crd = self.location_aux_coord.copy()
         additional_2d_crd.rename("kittens")
         self.diagnostic_cube_2d_time.add_aux_coord(additional_2d_crd, data_dims=(0, 1))
```
Comment (bayliffe) [inline improver_tests/spotdata/test_SpotExtraction.py:305]:
```suggestion
        self.diagnostic_cube_2d_aux.add_aux_coord(additional_2d_crd, data_dims=(0, 1))
```
Here and surrounds.

File: improver_tests/spotdata/test_SpotExtraction.py [hunk 7]
```diff
@@ -293,15 +281,16 @@ def test_scalar_and_nonscalar_coords(self):
...
-            if coord.name()
-            in ["time_in_local_timezone", "forecast_reference_time", "forecast_period"]
+            if coord.name() in ["time", "forecast_reference_time", "forecast_period"]
         ]
         expected_nonscalar = [self.expected_spot_time_coord]
+
         x_indices, y_indices = self.coordinate_cube.data
 
         scalar, nonscalar = plugin.get_aux_coords(
             self.diagnostic_cube_2d_time, x_indices, y_indices
```
Comment (bayliffe) [inline improver_tests/spotdata/test_SpotExtraction.py:291]:
```suggestion
            self.diagnostic_cube_2d_aux, x_indices, y_indices
```
You get the idea.

File: improver_tests/spotdata/test_SpotExtraction.py [hunk 8]
```diff
@@ -337,9 +326,8 @@ def test_coordinate_with_bounds_extraction(self):
         expected_points = self.expected_spot_time_coord.points
         expected_bounds = self.expected_spot_time_coord.bounds
         x_indices, y_indices = self.coordinate_cube.data
-
         points, bounds = plugin.get_coordinate_data(
-            self.diagnostic_cube_2d_time, x_indices, y_indices, coordinate="time"
+            self.diagnostic_cube_2d_time, x_indices, y_indices, coordinate="location"
```
Comment (bayliffe) [inline improver_tests/spotdata/test_SpotExtraction.py:330]:
```suggestion
            self.diagnostic_cube_2d_aux, x_indices, y_indices, coordinate="location"
```

File: improver_tests/spotdata/test_SpotExtraction.py [hunk 9]
```diff
@@ -353,9 +341,9 @@ def test_coordinate_without_bounds_extraction(self):
         expected_bounds = None
         x_indices, y_indices = self.coordinate_cube.data
 
-        self.diagnostic_cube_2d_time.coord("time").bounds = None
+        self.diagnostic_cube_2d_time.coord("location").bounds = None
```
Comment (bayliffe) [inline improver_tests/spotdata/test_SpotExtraction.py:344]:
```suggestion
        self.diagnostic_cube_2d_aux.coord("location").bounds = None
```

File: improver_tests/spotdata/test_SpotExtraction.py [hunk 10]
```diff
@@ -353,9 +341,9 @@ def test_coordinate_without_bounds_extraction(self):
...
         points, bounds = plugin.get_coordinate_data(
-            self.diagnostic_cube_2d_time, x_indices, y_indices, coordinate="time"
+            self.diagnostic_cube_2d_time, x_indices, y_indices, coordinate="location"
```
Comment (bayliffe) [inline improver_tests/spotdata/test_SpotExtraction.py:346]:
```suggestion
            self.diagnostic_cube_2d_aux, x_indices, y_indices, coordinate="location"
```

File: improver_tests/spotdata/test_SpotExtraction.py [hunk 11]
```diff
@@ -385,7 +373,7 @@ def test_building_cube(self):
             result.coord(self.unique_site_id_key).points, self.unique_site_id
         )
         self.assertArrayEqual(
-            result.coord("time").points, self.expected_spot_time_coord.points
+            result.coord("location").points, self.expected_spot_time_coord.points
```
Comment (bayliffe) [inline improver_tests/spotdata/test_SpotExtraction.py:376]:
```suggestion
            result.coord("location").points, self.expected_spot_aux_coord.points
```

File: improver_tests/spotdata/test_SpotExtraction.py [hunk 12]
```diff
@@ -489,12 +477,13 @@ def test_2d_aux_coords(self):
         retained as 1D coordinates associated with the spot-index on the
         spotdata cube."""
         plugin = SpotExtraction()
+        print(self.diagnostic_cube_2d_time)
```
Comment (bayliffe) [inline improver_tests/spotdata/test_SpotExtraction.py:480]: Delete.

File: improver_tests/spotdata/test_SpotExtraction.py [hunk 13]
```diff
@@ -489,12 +477,13 @@ def test_2d_aux_coords(self):
...
         result = plugin.process(
             self.neighbour_cube,
             self.diagnostic_cube_2d_time,
             new_title="IMPROVER Spot Forecast",
         )
-        self.assertEqual(result.coord("time"), self.expected_spot_time_coord)
+        self.assertEqual(result.coord("location"), self.expected_spot_time_coord)
```
Comment (bayliffe) [inline improver_tests/spotdata/test_SpotExtraction.py:486]:
```suggestion
        self.assertEqual(result.coord("location"), self.expected_spot_aux_coord)
```

**Review by bayliffe — CHANGES_REQUESTED** [review]
Thanks again. Tiny stuff left.

File: improver/developer_tools/metadata_interpreter.py (hunk 1 above)
Comment (mspelman07) [reply improver/developer_tools/metadata_interpreter.py]: Quite right. I've removed it entirely.

File: improver_tests/spotdata/test_SpotExtraction.py (hunk 2 above)
Comment (mspelman07) [reply improver_tests/spotdata/test_SpotExtraction.py]: updated

Comment (mspelman07) [reply improver_tests/spotdata/test_SpotExtraction.py]: updated everywhere.

Comment (mspelman07) [reply improver_tests/spotdata/test_SpotExtraction.py]: updated everywhere

File: improver_tests/spotdata/test_SpotExtraction.py (hunk 4 above)
Comment (mspelman07) [reply improver_tests/spotdata/test_SpotExtraction.py]: updated everywhere

File: improver_tests/spotdata/test_SpotExtraction.py (hunk 5 above)
Comment (mspelman07) [reply improver_tests/spotdata/test_SpotExtraction.py]: Deleted remaining errant print statements.

**Review by bayliffe — CHANGES_REQUESTED** [review]
Marcus, you just need to recreate the acceptance test data checksums following the removal of the spot-extract data. Otherwise this is now fine.

**Comment by mspelman07** [comment]
I'd actually just forgotten to commit the acceptance test data changes around the spot-extract data (but the test were all passing for me so I hadn't noticed). It should all work now if you update the acceptance test branch.

**Review by MoseleyS — CHANGES_REQUESTED** [review]
Looking good. I think I've found one reference in the documentation that needs reviewing.

**Comment by mspelman07** [comment]
I found one piece of documentation that needed removing. Hopefully it's the same one you were thinking of?

**Review by MoseleyS — CHANGES_REQUESTED** [review]
You found the right bit. I'm not sure what happened to my comment, it seems to have been lost somewhere along the way.

`numba` and `pytz` now don't have a "Required for CLIs" section. This makes `pytz` seem redundant, but I think there is somewhere it is still used. Can you reference it here. I guess `numba` still does some good, although I know we can't use it on our current HPC.

**Comment by mspelman07** [comment]
From a bit of digging I found that pytz was actually only being used by one file that could be changed to use datetime's timezone so I've changed it now and removed pytz from the environments.
 
numba is used by the ensemble copula coupling file if it is available for faster computation of the interpolations. I've tried to track down the CLI's using this and added them to the dependencies file.

**Review by MoseleyS — APPROVED** [review]
RIP Timezone extraction

**Pull request merged**